scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -149,252 +196,374 @@ sce_list
-
 Now that we have a list of processed SCE objects, we need to merge the objects into one overall SCE object which we can provide as input to integration methods.
 A word of caution before we begin: **This merged SCE object is NOT an integrated SCE!** 
+Merging SCEs itself does not perform any batch correction; that's why we proceed to integration next.
 
 To merge SCE objects, we do need to do some wrangling and bookkeeping to ensure compatibility and that we don't lose important information.
-Overall we'll need to take care of these items: 
+Overall we'll want to take care of these items: 
 
-- We should be able to trace library-specific information back to its originating library, including...
-  - Library-specific statistics, e.g. gene-level statistics for a given library found in `rowData`
+1. We should be able to trace library-specific information back to the originating library, including...
   - Cell-level information: Which library is each cell from? 
   Although cells have unique barcode identifiers, these barcodes are only guaranteed to be unique _within_ a library. 
-  Now that we are combining libraries, there's no guarantee we can identify a cell's originating library from its barcode alone.
-- SCE objects should contain the same genes. In other words, each `rowData` table should have the same rownames.
-- SCE columns should match. In other words, the each `colData` should have the same column names.
+  Now that we are merging several libraries, we won't necessarily be able to identify a cell's originating library from its barcode alone.
+  - Library-specific statistics, e.g. gene-level statistics for a given library found in `rowData`
+2. SCE objects should contain the same genes: Each SCE object should have the same row names.
+3. SCE cell metadata columns should match: Each SCE object should have the same column names.
+
+Here is a schematic of what how one of the SCE objects will ultimately be modified into the final merged SCE:
+
+![](diagrams/technical_merge_sce.png)
+
+We'll begin by take some time to thoroughly explore our SCE objects to figure out what wrangling steps we need to take for these specific data.
+Don't skip this exploration!
+Bear in mind that the exact wrangling shown here will not be the same for other SCE objects you work with, but the same general principles apply.
+
+
+#### Preserving library information at the cell level
 
+How will we be able to tell which library a given cell came from?
 
-First, let's determine what the shared genes among SCE objects are using some `purrr` magic.
-Ultimately, we'll filter SCEs down to contain only these genes, thereby ensuring that their genes match.
+First, we'll want to directly add information to the SCE that indicates the different libraries.
+The best way to do this is simply to add a `colData` column with the library information.
```

Comment:
```suggestion
We must ensure that the column names of the SCE object are unique and directly add information to the SCE that indicates the different libraries.
The best way to do this is simply to add a `colData` column with the library information.
```
I think the punchline here is two fold - there must be a column within the metadata that says where each cell comes from AND the column names of the SCE object must be unique. I think the column names being unique is more important here and isn't an optional step so that should be stated first and the second paragraph here needs to be slightly reworded, but feel free to disagree with me.

Reply:
fine with me!